### PR TITLE
[WIP] ThemedSettings (the easy way)

### PR DESCRIPTION
**Pull request by vanitasvitae**
There are still some bugs left to remove, but all in all it works already

TODO:
- [x] Make sure, nothing breaks, when updating the app from 0.1.6 to this
- [x] Reimplement color circle previews

### CODE REVIEW

File: app/src/main/java/com/github/dfa/diaspora_android/activity/SettingsActivity.java [hunk 1]
```diff
@@ -64,153 +51,142 @@
...
     private ProxyHandler.ProxySettings oldProxySettings;
 
-
-    @Override
-    public void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-        setContentView(R.layout.settings__activity);
+    public void onCreate(Bundle b) {
+        super.onCreate(b);
+        this.setContentView(R.layout.settings__activity);
```
Comment (gsantner) [inline app/src/main/java/com/github/dfa/diaspora_android/activity/SettingsActivity.java]: this. not needed

File: app/src/main/java/com/github/dfa/diaspora_android/activity/SettingsActivity.java [hunk 2]
```diff
@@ -64,153 +51,142 @@
...
+                    fragment = new SettingsFragmentDebugging();
+                    break;
+                case SettingsFragmentMaster.TAG:
+                default:
+                    fragment = new SettingsFragmentMaster();
+                    break;
+            }
         }
+        FragmentTransaction t = getFragmentManager().beginTransaction();
+        if(addToBackStack) t.addToBackStack(tag);
```
Comment (gsantner) [inline app/src/main/java/com/github/dfa/diaspora_android/activity/SettingsActivity.java]: if () { 

}

File: app/src/main/java/com/github/dfa/diaspora_android/activity/SettingsActivity.java [hunk 3]
```diff
@@ -64,153 +51,142 @@
...
+                ProxyHandler.getInstance().updateProxySettings(this);
+            }
+        }
+        super.onStop();
     }
 
-    public static class SettingsFragment extends PreferenceFragment implements SharedPreferences.OnSharedPreferenceChangeListener {
-        private SharedPreferences sharedPreferences;
+    public static class SettingsFragmentMaster extends ThemedPreferenceFragment {
+        public static final String TAG = "com.github.dfa.diaspora_android.SettingsFragmentMaster";
```
Comment (gsantner) [inline app/src/main/java/com/github/dfa/diaspora_android/activity/SettingsActivity.java]: Note in review descprition

File: app/src/main/res/xml/preferences__master.xml [hunk 4]
```diff
@@ -0,0 +1,110 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (gsantner) [inline app/src/main/res/xml/preferences__master.xml:1]: Really nice that it works this way! :+1: :D

**Review by gsantner — COMMENTED** [review]
Really great work :+1: . I'm very happy that it can be done (and was done) this way :D.

TAG = "com.github.dfa.diaspora_android.SettingsFragment*"
--> com.github.dfa.diaspora_android.SOMEPREFIX.SettingsFragmentMaster
Ideas SOMEPREFIX = .fragment. ? .settings. ? .ui.?

File: app/src/main/java/com/github/dfa/diaspora_android/activity/SettingsActivity.java (hunk 1 above)
Comment (vanitasvitae) [reply app/src/main/java/com/github/dfa/diaspora_android/activity/SettingsActivity.java]: Fixed

File: app/src/main/java/com/github/dfa/diaspora_android/activity/SettingsActivity.java (hunk 3 above)
Comment (vanitasvitae) [reply app/src/main/java/com/github/dfa/diaspora_android/activity/SettingsActivity.java]: Done

File: app/src/main/res/xml/preferences__master.xml (hunk 4 above)
Comment (vanitasvitae) [reply app/src/main/res/xml/preferences__master.xml:1]: I know right? :D

**Comment by vanitasvitae** [comment]
I think this is now ready to be merged :dancer:

File: app/src/main/res/drawable/circle.xml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="oval">
+    <solid android:color="@color/md_red_500"/>
+    <size android:width="@dimen/color_picker_circle_size" android:height="@dimen/color_picker_circle_size"/>
```
Comment (vanitasvitae) [inline app/src/main/res/drawable/circle.xml:4]: That pity line took me like 1 hour to debug, until I noticed, that it was missing :angry:

**Pull request merged**